spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindReturnRef.java

```diff
@@ -80,13 +84,16 @@ public class FindReturnRef extends OpcodeStackDetector {
     private final Map<OpcodeStack.Item, OpcodeStack.Item> arrayParamClones = new HashMap<>();
     private final Map<XField, List<OpcodeStack.Item>> fieldValues = new HashMap<>();
 
+    private final Map<XMethod, XField> accessMethodFields = new HashMap<>();
+
     private final BugAccumulator bugAccumulator;
 
     private static final Matcher BUFFER_CLASS_MATCHER = Pattern.compile("Ljava/nio/[A-Za-z]+Buffer;").matcher("");
     private static final Matcher DUPLICATE_METHODS_SIGNATURE_MATCHER =
             Pattern.compile("\\(\\)Ljava/nio/[A-Za-z]+Buffer;").matcher("");
     private static final Matcher WRAP_METHOD_SIGNATURE_MATCHER =
             Pattern.compile("\\(\\[.\\)Ljava/nio/[A-Za-z]+Buffer;").matcher("");
+    private static final Matcher ACCESS_METHOD_MATCHER = Pattern.compile("access\\$\\d{3}").matcher("");
```

Comment:
`Matcher` instances are not thread-safe. This should store the `Pattern` and then create a `Matcher` for each string, not reset the same Matcher.